azext_iot/tests/digitaltwins/test_dt_resource_lifecycle_int.py

```diff
@@ -370,6 +378,276 @@ def test_dt_endpoints_routes(self):
 
         eventgrid_endpoint = "myeventgridendpoint"
 
+        logger.debug("Adding key based eventgrid endpoint...")
+        self.cmd(
+            "dt endpoint create eventgrid -n {} -g {} --egg {} --egt {} --en {} --dsu {} --no-wait".format(
+                endpoints_instance_name,
+                self.rg,
+                EP_RG,
+                EP_EVENTGRID_TOPIC,
+                eventgrid_endpoint,
+                MOCK_DEAD_LETTER_SECRET,
+            )
+        )
+        endpoint_tuple_collection.append(
+            EndpointTuple(
+                eventgrid_endpoint,
+                ADTEndpointType.eventgridtopic,
+                ADTEndpointAuthType.keybased,
+            )
+        )
+
+        servicebus_endpoint = "myservicebusendpoint"
+        servicebus_endpoint_msi = "{}identity".format(servicebus_endpoint)
+
+        logger.debug("Adding key based servicebus topic endpoint...")
+        self.cmd(
+            "dt endpoint create servicebus -n {} --sbg {} --sbn {} --sbp {} --sbt {} --en {} --dsu {} --no-wait".format(
+                endpoints_instance_name,
+                EP_RG,
+                EP_SERVICEBUS_NAMESPACE,
+                EP_SERVICEBUS_POLICY,
+                EP_SERVICEBUS_TOPIC,
+                servicebus_endpoint,
+                MOCK_DEAD_LETTER_SECRET,
+            )
+        )
+        endpoint_tuple_collection.append(
+            EndpointTuple(
+                servicebus_endpoint,
+                ADTEndpointType.servicebus,
+                ADTEndpointAuthType.keybased,
+            )
+        )
+
+        logger.debug("Adding identity based servicebus topic endpoint...")
+        self.cmd(
+            "dt endpoint create servicebus -n {} --sbg {} --sbn {} --sbt {} --en {} --du {} "
```

Comment:
Ensuring endpoints can be created/updated in both key or identity auth modes can happen in the endpoint tests rather than the endpoint route tests.

Reply:
updating endpoints is something we have not advertised to customers and running create on an existing endpoint seems to have behavior that we do not expect. Will not test for update (or expose to customers) for now.